EIPS/eip-3779.md

```diff
@@ -44,69 +41,28 @@ The relative jumps of [EIP-4200](./eip-4200) and the simple subroutines of [EIP-
 > `RJUMPSUB` offset
 * Pushes _IP+1_ on the return stack and jumps to _IP+offest_.
 > `RETURNSUB`
-* Jumps to the address popped off the return stack.
+* Pops an address off the return stack and jumps to it,
 
 Note that each jump creates at most two paths of control through the code, such that the complexity of traversing the entire control-flow graph is linear in the size of the code.
 
 #### *Dynamic Jumps*
 
-Dynamic jumps, where the destination of a `JUMP` or `JUMPI` is not known until runtime, are an obstacle to proving validity in linear time -- any jump can be to any destination in the code, potentially requiring time quadratic in the size of code. For this reason we have two real choices.
-
-1. Deprecate dynamic jumps. This is easily done:
-
-> Define `JUMP` and `JUMPI` as `INVALID` for the purposes of [EIP-3670: EOF Code Validation](./eip-3670.md).
-
-2. Constrain dynamic jumps. This requires static analysis.
-
-Consider the simplest and most common case.
-```
-PUSH address
-JUMP
-```
-This is effectively a static jump.
-
-Another important use of `JUMP` is to implement the return jump from a subroutine. So consider this example of calling and returning from a minimal subroutine:
-```
-TEST_SQUARE:
-    jumpdest
-    push RTN_SQUARE 
-    0x02
-    push SQUARE
-    jump
-RTN_SQUARE
-    jumpdest
-    swap1
-    jump
-
-SQUARE:
-    jumpdest
-    dup1
-    mul
-    swap1
-    jump
-```
-The return address -`RTN_SQUARE` - and the destination address - `SQUARE` - are pushed on the stack as constants and remain unchanged as they move on the stack, such that only those constants are passed to each `JUMP`. They are effectively static.  We can track the motion of constants on the `data stack` at validation time, so *we do not need unconstrained dynamic jumps to implement subroutines.*
-
-*The above is the simplest analysis that suffices.  A more powerful analysis that takes in more use cases is possible -- slower, but still linear-time.*
+`JUMP` or `JUMPI` also provide a complete set of control flow instructions.  But they are also an obstacle to proving validity in linear time -- jumps can be to any destination in the code, creating a "path explosion" potentially requiring time quadratic in the size of code.  We propose to deprecate them.
 
 #### Validation
 
-We can validate the safety of contracts with a static analysis that takes time and space linear in the size of the *code*, as shown below.  And since we can, we should.
+We can validate contracts with a static analysis that takes time and space linear in the size of the *code*, as shown below.  And since we can, we should.
```

Comment:
```suggestion
We can validate contracts with a static analysis that takes time and space linear in the size of the *code*, as shown below. And since we can, we should.
```
Also: no first person.